lock/lock.go

```diff
@@ -108,19 +108,26 @@ func (d *Driver) ExecuteWithMultiUseLock(ucs []config.UseLocker, timeout time.Du
 // lock after a timeout, returning a stop channel. Timeout is jittered to
 // mitigate thundering herd problems.
 func (d *Driver) AcquireWithTTLRefresh(uc config.UseLocker, ttl, timeout time.Duration) (chan struct{}, error) {
+	if err := d.Config.PublishUse(uc); err != nil {
+		return nil, err
+	}
+
 	if err := d.acquire(uc, ttl, timeout); err != nil {
 		return nil, err
 	}
 
-	stopChan := make(chan struct{})
+	stopChan := make(chan struct{}, 1)
```

Comment:
I understand the implication of this change (the first send on the channel will not block now), but what was the reason for the change?